IM_calculation/IM/im_calculation.py

```diff
@@ -413,28 +435,35 @@ def compute_measures_multiprocess(
     ) = constants.Components.get_comps_to_calc_and_store(comp)
 
     bbseries, station_names = get_bbseis(input_path, file_type, station_names)
-
     total_stations = len(station_names)
+    # determine the size of each iteration base on num of processers and mem
     steps = get_steps(
         input_path, process, total_stations, "FAS" in ims and bbseries.nt > 32768
     )
 
     all_results = []
     p = Pool(process)
 
+    if advanced_im_config.IM_list:
+        df_adv_im = {im: pd.DataFrame() for im in advanced_im_config.IM_list}
+
     i = 0
     while i < total_stations:
```

Comment:
Not requiring a change, but why is this a while loop instead of "for i in range(0, total_stations, steps)" loop?

Reply:
Because our code scrutiny wasn't as high as it is now when this code was written.

Reply:
i'm not sure whoever wrote the first version of this, but i personally like while better than for. i think its easier to follow the logic and more flexible ( instead of set sequence, you can do more complex manipulation)

Reply:
Imo the only time you should write whiles in python is when you don't know how many iteration steps you are going to take.

Reply:
:+1:  to Jason, I'm in favour of this being a for loop *eventually*. Not for this PR though

Reply:
Imo the only time you should write whiles in python is when you don't know how many iteration steps you are going to take. Aka iterating until a certain condition is met. There's a lot of potential for bugs with while loops.